crates/vm/backends/levm/mod.rs

```diff
@@ -195,16 +196,32 @@ impl LEVM {
             difficulty: block_header.difficulty,
         };
 
-        let mut vm = VM::new(
-            tx.to(),
-            env,
-            tx.value(),
-            tx.data().clone(),
-            db.clone(),
-            block_cache.clone(),
-            tx.access_list(),
-            tx.authorization_list(),
-        )?;
+        let mut vm = if let Transaction::PrivilegedL2Transaction(privileged_tx) = tx {
```

Comment:
Can't we make this `if` statement happen inside the hook? That is, can't we have just one L2 hook that we use for everything and make the hook inside decide which validations to perform depending on whether the transaction is a privileged one or not?

Reply:
Check [1c273fe](https://github.com/lambdaclass/ethrex/pull/2332/commits/1c273fecf07f624835971cfac89fd7a5d6f5ca3d)